### ISSUE TITLE
Comparison between LambdaRank output scores

### ISSUE THREAD

**Issue by dingyf05** [microsoft/LightGBM#5585]
The trained lambdarank model outputs score like [0.2, 0.5, -0.1], which can get rank result by sort the prediction score (in per query) .
I wonder if the prediction scores are relatively comparable between different queries? For example if one item with score 0.2 in query1 ranks higher than one with score 0.1 in query2?

**Comment by shiyu1994** [comment]
> I wonder if the prediction scores are relatively comparable between different queries? For example if one item with score 0.2 in query1 ranks higher than one with score 0.1 in query2?

Thanks for using LightGBM.

The short answer is no. In lambdarank objective, the gradients and hessians are calculated per-query. That means, the score and label of an item only affects the calculation of gradients and hessians of items in the same query. So the prediction values of items in different queries is not comparable.
